service/test_automatic_run_experiment.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for automatic_run_experiment.py"""
+import os
+from unittest import mock
+
+from common import utils
+from service import automatic_run_experiment
+
+# pylint: disable=invalid-name,unused-argument
+
+
+@mock.patch('experiment.run_experiment.start_experiment')
+@mock.patch('experiment.stop_experiment.stop_experiment')
+@mock.patch('src_analysis.experiment_changes.get_fuzzers_changed_since_last')
+@mock.patch('service.automatic_run_experiment.get_experiment_name')
+def test_run_diff_experiment(mocked_get_experiment_name,
+                             mocked_get_fuzzers_changed_since_last,
+                             mocked_stop_experiment, mocked_start_experiment,
+                             db):
+    """Tests that run_diff_experiment starts and stops the experiment
+    properly."""
+    expected_experiment_name = 'myexperiment'
+    mocked_get_experiment_name.return_value = expected_experiment_name
+    fuzzers = ['afl', 'aflplusplus']
+    mocked_get_fuzzers_changed_since_last.return_value = fuzzers
+    automatic_run_experiment.run_diff_experiment(False)
+    expected_config_file = os.path.join(utils.ROOT_DIR, 'service',
+                                        'experiment-config.yaml')
+
+    def sort_key(dictionary):
+        return dictionary['fuzzer']
+
+    expected_fuzzer_configs = list(
+        sorted([{
+            'fuzzer': fuzzer
+        } for fuzzer in fuzzers], key=sort_key))
+    expected_benchmarks = [
```

Comment:
Will this list have to be changed as benchmarks change ? Better to do something else or call get_all_benchmarks function ?

Reply:
It will but we will also need to update the list in automatic_run_experiment so I think it's fine. I can use `automatic_run_experiment.BENCHMARKS` so that less updating is required, but i tried to follow the philosophy of relying as little on production code as possible when testing.